File: src/main/java/de/gieskerb/class_craft/listener/EntityBreedListener.java
package de.gieskerb.class_craft.listener;

import de.gieskerb.class_craft.classes.BaseClass;
import de.gieskerb.class_craft.classes.FarmerClass;
import de.gieskerb.class_craft.data.PlayerData;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityBreedEvent;

public class EntityBreedListener implements Listener {

    @EventHandler
    public void onEntityBreed(EntityBreedEvent event) {
        if (!(event.getBreeder() instanceof Player player)) {
            return;
        }

        BaseClass playerClass = PlayerData.getPlayerData(player.getName()).getActiveClass();
        if (playerClass instanceof FarmerClass farmerClass) {
            player.sendMessage("You have breed an animal!");
        }
    }

}

File: src/main/kotlin/de/gieskerb/class_craft/listener/EntityBreedListener.kt
package de.gieskerb.class_craft.listener

import de.gieskerb.class_craft.classes.BaseClass
import de.gieskerb.class_craft.classes.FarmerClass
import de.gieskerb.class_craft.data.PlayerData
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityBreedEvent

class EntityBreedListener : Listener {
    @EventHandler
    fun onEntityBreed(event: EntityBreedEvent) {
        if (event.breeder !is Player) {
            return
        }
        val player: Player = event.breeder as Player;
        val playerClass: BaseClass? = PlayerData.getPlayerData(player.name)?.activeClass
        if (playerClass is FarmerClass) {
            player.sendMessage("You have breed an animal!")
        }
    }
}